### PR TITLE
fix the bug where DnD from projection column with a special lambda name generates wrong pure code

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

fix the bug where DnD from projection column with a special lambda name generates wrong pure code

## How did you test this change?

- [x] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

before: 

https://github.com/finos/legend-studio/assets/73408381/9d368e20-4215-46ae-928e-471ab94368c6

after: 

https://github.com/finos/legend-studio/assets/73408381/15b0a359-c2df-49d4-8723-dd0358e29784

### CODE REVIEW

File: packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx [hunk 1]
```diff
@@ -247,9 +273,9 @@ export const buildFilterTreeWithExists = (
       const parentPropertyChainIndex = existsLambdaPropertyChains.findIndex(
         (p) =>
           p instanceof AbstractPropertyExpression &&
-          p.func.value ===
-            targetDropNode.propertyExpressionState.propertyExpression.func
-              .value &&
+          p.func.value.hashCode ===
+            targetDropNode.propertyExpressionState.propertyExpression.func.value
+              .hashCode &&
```
Comment (gayathrir11) [inline packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:278]: I don't think we should use hashcode to check the equality. Do you know where the inequality is happening. I remember that it used to work without deepclone

Comment (YannanGao-gs) [reply packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:278]: Here we are comparing two Property. they have same values but different reference.

Comment (YannanGao-gs) [reply packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:278]: hashcode is generated upon  👀

Comment (gayathrir11) [reply packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:278]: Got it. I just feel hashcode is something we use to do change detection. We generally don't use it do these kinds of equality checks across the codebase If deepClone is failing this check to preserve the reference of property we can try using `buildPropertyExpressionChain` function to build the new expression. This was you hold no references to the property expression present in project. @MauricioUyaguari  what do you think?

**Pull request closed without merging**